plugins/circleci/src/circleci-config.ts

```diff
@@ -196,8 +198,10 @@ export default abstract class CircleCiConfigHook extends Hook<CircleCIState> {
   _versionTag?: string
   abstract config: CircleCIStatePartial
 
-  async getCircleConfig(): Promise<string | undefined> {
+  async getCircleConfig(): Promise<[string | undefined, boolean]> {
+    let firstLoad = false
```

Comment:
**nitpick:** This to me reads as _this is not the first load_ whereas we want to say _the first load hasn't happened yet_. To me I would name this `hasCheckedBaseConfig` to be super verbose matching the intent of this Boolean.

Reply:
I'm confused, it _does_ mean "this is not the first load", which is how you read it. Are you saying you want it to mean "the first load hasn't happened yet"?

Reply:
Yes 👍 To me - as soon as `getCircleConfig()` is called, that is the first load so `firstLoad` should be true. I was not able to figure out the purpose/intent of `firstLoad` without parsing the code in my head. My aim for the nitpick is to make the variable name clearer so I can now its' intent without having to read the whole function and how it is used.

Reply:
I'd argue that we shouldn't change the variable to something like `hasCheckedBaseConfig` as `getCircleConfig` shouldn't concern itself with what callees are doing with the `firstLoad` (or whatever) boolean – it's a leaky abstraction, and might no longer be correctly named if we later add another piece of logic that uses the boolean.

Reply:
Is there anyway we can make this clearer? I struggled to understand its' intent when first reading the block of code. The aim here is to make this easier to maintain.

Reply:
`isFirstLoad`? `isFirstLoadOfConfig`? I'm still struggling to understand what's unclear about it, sorry!

> as soon as getCircleConfig() is called, that is the first load so firstLoad should be true

this isn't true the majority of times it's called, only the first time it's called 😅